test-cloud/test-cloud-common/src/main/java/org/kie/cloud/tests/common/client/util/WorkbenchUtils.java

```diff
@@ -33,28 +40,60 @@
+    public static void deployProjectToWorkbench(String repositoryName,
+                                                KieDeploymentScenario<?> deploymentScenario,
+                                                String projectName) {
         GitProvider gitProvider = deploymentScenario.getGitProvider();
         WorkbenchDeployment workbenchDeployment = deploymentScenario.getWorkbenchDeployments().get(0);
 
         deployProjectToWorkbench(gitProvider.getRepositoryUrl(repositoryName), workbenchDeployment, projectName);
     }
 
-    public static void deployProjectToWorkbench(String repositoryUrl, WorkbenchDeployment workbenchDeployment, String projectName) {
+    public static void deployProjectToWorkbench(String repositoryUrl,
+                                                WorkbenchDeployment workbenchDeployment,
+                                                String projectName) {
         CloneProjectRequest cloneProjectRequest = createCloneProjectRequest(repositoryUrl, projectName);
 
         WorkbenchClient workbenchClient = WorkbenchClientProvider.getWorkbenchClient(workbenchDeployment);
         workbenchClient.createSpace(SPACE_NAME, workbenchDeployment.getUsername());
-        workbenchClient.cloneRepository(SPACE_NAME, cloneProjectRequest);
+        for (int tries = 0; tries < 5; tries++) {
+            try {
+                logger.info("Cloning project. Try: " + tries);
+                workbenchClient.cloneRepository(SPACE_NAME, cloneProjectRequest);
+                logger.info("Project was cloned to the workbench after " + (tries + 1) + " attempts.");
+                break;
+            } catch (ClientRequestTimedOutException ex) {
+                logger.warn("Caught exception during cloning repository to the Workbench. Waiting for 5 minutes to see if the project was created.", ex);
+                try {
+                    new SimpleWaiter(() -> workbenchClient.getProjects(SPACE_NAME).stream().map(
+                                                                                                ProjectResponse::getName)
+                                                          .anyMatch(projectName::equals)).reason("Waiting for "
+                                                                  + projectName + " to be cloned into Workbench")
+                                                                                         .timeout(TimeUnit.MINUTES, 2)
+                                                                                         .interval(TimeUnit.SECONDS, 5)
+                                                                                         .waitFor();
+                    logger.info("Container after cloning timeout found");
+                } catch (NotFoundException e) {
+                    handleNotFoundException(workbenchClient, e);
+                } catch (WaiterException e) {
+                    handleWaiterException(tries, e);
+                }
+            }
+        }
```

Comment:
I think this logic should be part of the cloneRepository method. 
Also, as the previous comment, I would prefer using the AwaitilityUtils for retries.

Reply:
Don't you think that ensuring the repository is created in any kind of environments and methods is a good thing? In the best case (non HA environments) we will only lose few time. Also, we could spot issues earlier (in case of for some reason, the repository is not created). This is just my opinion.